### PR TITLE
fix(platform): fix date picker accessibility issue

### PR DESCRIPTION
**Pull request by Betrozov**
#### Please provide a link to the associated issue.
Closes SAP/fundamental-ngx#5080

#### Please provide a brief summary of this pull request.
Fixed issue №1 from the doc: the information `required` should be read for all the mandatory (*) marked fields on navigation
Fixed issue №4 from the doc: add a mouse hover tooltip to date picker buttons

this PR was already reviewed and approved here:
https://github.com/SAP/fundamental-ngx/pull/5572
https://github.com/SAP/fundamental-ngx/pull/5546

#### Please check whether the PR fulfills the following requirements

- [X] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [X] tests for the changes that have been done
- [X] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
- [X] Run npm run build-pack-library and test in external application

Documentation checklist:
- [X] update `README.md`
- [X] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [X] Documentation Examples
- [X] Stackblitz works for all examples

### CODE REVIEW

**Review by olshansky — APPROVED** [review]

File: apps/docs/src/app/platform/component-docs/platform-forms/platform-date-picker/platform-date-picker-examples/platform-date-picker-example.component.html [hunk 1]
```diff
@@ -27,7 +28,7 @@ <h3>Date Picker with Reactive Form.</h3>
     <fdp-form-field #ffl5 id="disableddate" column="1" rank="2" required="true" label="Disabled Date Picker:"
         disabled="true">
         <fdp-date-picker name="disableddate" type="single" [allowNull]="false"
-            placeholder="This is a disabled date picker" [formControl]="ffl5.formControl">
+            placeholder="This is a disabled date picker" [formControl]="ffl5.formControl" ariaLabel="required">
```
Comment (DeepakSap14) [inline apps/docs/src/app/platform/component-docs/platform-forms/platform-date-picker/platform-date-picker-examples/platform-date-picker-example.component.html:31]: https://github.com/SAP/fundamental-ngx/pull/5627 is fixing reading of required field for date-picker as well.
adding ariaLabel ="required" is not correct solution.
please see https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-required_attribute.

**Review by InnaAtanasova — APPROVED** [review]

**Review by mikerodonnell89 — CHANGES_REQUESTED** [review]
See @DeepakSap14 's comment.  also since html5 screenreaders will correctly read off simply using the `required` attribute. [link](https://www.w3.org/WAI/GL/wiki/Techniques/HTML5/Using_the_required_attribute_to_indicate_a_required_input_field#:~:text=The%20HTML%205%20required%20attribute,readers%20identify%20it%20as%20required.)

**Comment by DeepakSap14** [comment]
one request:
since we are fixing date-picker accessibility, it can include fix for reading the label for date picker.
currently because of [code-line](https://github.com/SAP/fundamental-ngx/blob/0293a0a18ec740e85a681ffda4d32e20df898e10/libs/core/src/lib/date-picker/date-picker.component.html#L21)
label is not being read.
It will fix this issue in core and platform both.
same issue is also present in date-time-picker as well.

**Review by dimamarksman — APPROVED** [review]

File: apps/docs/src/app/platform/component-docs/platform-forms/platform-date-picker/platform-date-picker-examples/platform-date-picker-example.component.html [hunk 2]
```diff
@@ -24,7 +24,7 @@ <h3>Date Picker with Reactive Form.</h3>
             [formControl]="ffl4.formControl">
         </fdp-date-picker>
     </fdp-form-field>
-    <fdp-form-field #ffl5 id="disableddate" column="1" rank="2" label="Disabled Date Picker:"
+    <fdp-form-field #ffl5 id="disableddate" column="1" rank="2" required="true" label="Disabled Date Picker:"
```
Comment (mikerodonnell89) [inline apps/docs/src/app/platform/component-docs/platform-forms/platform-date-picker/platform-date-picker-examples/platform-date-picker-example.component.html:27]: It is better to simply use `required` without `="true"` or `="false"` - the presence of the required property will always evaluate to true: https://stackoverflow.com/a/48651305/2217983 

```suggestion
    <fdp-form-field #ffl5 id="disableddate" column="1" rank="2" required label="Disabled Date Picker:"
```

**Review by mikerodonnell89 — CHANGES_REQUESTED** [review]
Looks good just one small thing

**Pull request closed without merging**